### ISSUE TITLE
Maximum call stack size exceeded with Babel

### ISSUE THREAD

**Issue by tomekbuszewski** [fuse-box/fuse-box#42]
Hi,

I wanted to make a switch from Webpack to FuseBox with Gulp. My previous .babelrc file looked like this:
```
{
  "presets": ["es2015", "stage-0", "react"]
}

```
So I though I want this also to be in my FuseBox config. My gulpfile is as follows:
```
var fuse = new fusebox.FuseBox({
  homeDir:   'app/js',
  sourceMap: {
    bundleReference: "index.js.map",
    outFile:         "./dist/index.js.map",
  },
  cache:     true,
  outFile:   'dist/index.js',
  plugins:   [
    fusebox.BabelPlugin({
      config: {
        sourceMaps: true,
        presets: ["es2015", "stage-0", "react"],
        plugins: [
          ["transform-react-jsx"]
        ]
      }
    })
  ]
});

gulp.task('js', function () {
  return fuse.bundle('>index.js');
});
```

When I run `gulp js`, I am getting
```
➜  lesnik git:(fuse) ✗ gulp js
[11:34:50] Using gulpfile ~/Sites/lesnik/gulpfile.js
[11:34:50] Starting 'js'...
RangeError: Maximum call stack size exceeded
    at pre (/Users/tomekbuszewski/Sites/lesnik/node_modules/fuse-box/dist/commonjs/FileAnalysis.js:53:18)
    at visit (/Users/tomekbuszewski/Sites/lesnik/node_modules/ast-traverse/ast-traverse.js:16:19)
    at visit (/Users/tomekbuszewski/Sites/lesnik/node_modules/ast-traverse/ast-traverse.js:32:21)
    at visit (/Users/tomekbuszewski/Sites/lesnik/node_modules/ast-traverse/ast-traverse.js:32:21)
    at visit (/Users/tomekbuszewski/Sites/lesnik/node_modules/ast-traverse/ast-traverse.js:32:21)
    at visit (/Users/tomekbuszewski/Sites/lesnik/node_modules/ast-traverse/ast-traverse.js:32:21)
    at visit (/Users/tomekbuszewski/Sites/lesnik/node_modules/ast-traverse/ast-traverse.js:32:21)
    at visit (/Users/tomekbuszewski/Sites/lesnik/node_modules/ast-traverse/ast-traverse.js:32:21)
    at visit (/Users/tomekbuszewski/Sites/lesnik/node_modules/ast-traverse/ast-traverse.js:29:25)
    at visit (/Users/tomekbuszewski/Sites/lesnik/node_modules/ast-traverse/ast-traverse.js:29:25)
```

I tried using preset `latest`, but then spread operators are mistreated:
```
[0] [11:31:08] Starting 'js'...
[0] SyntaxError: unknown: Unexpected token (9:15)
[0]    7 |   switch (action.type) {
[0]    8 |     case 'ADD_POST':
[0] >  9 |       return { ...state, posts: [ ...state.posts, action.payload ] };
```

I am kinda stuck with this one. Any ideas?

**Comment by nchanged** [comment]
Hi!
Which version are you using?
Could you provide a sample code so we can reproduce it?

**Comment by nchanged** [comment]
I think i am able to reproduce it now
```
let a = function(...args) {
    return {...args }
}
```

Will check what's wrong

**Comment by tomekbuszewski** [comment]
Hi,

the code you provided is similar to mine, spread operator is the problem here. Will wait for updates, and thanks for the quick response! ;)

**Comment by nchanged** [comment]
@tomekbuszewski no problem, I got the issue already, related to a circular AST structure provided by babel. And I am traversing it

**Comment by nchanged** [comment]
@tomekbuszewski Good news, i think i was able to fix it in `1.3.26`. You can update and verify the fix.

Cheers,

**Comment by tomekbuszewski** [comment]
It works, thanks @nchanged !
